src/Data/Singletons/Single/Monad.hs

```diff
@@ -228,3 +228,6 @@ singDecsM :: DsMonad q => [Dec] -> SgM [DDec] -> q [DDec]
 singDecsM locals thing = do
   (decs1, decs2) <- singM locals thing
   return $ decs1 ++ decs2
+
+proxyFor :: DType -> DExp
```

Comment:
You are right, I intended this function to be temporary, which is why it's in an inappropriate module and uses `Proxy` names this way. Looking forward to getting rid of it.